hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```diff
@@ -52,56 +82,595 @@ protected AbstractGrid(final VaadinMessageSource i18n, final UIEventBus eventBus
+            setHiddenColumns();
+            setColumnExpandRatio();
+        }
+
+        /**
+         * Sets the column properties for maximized-state.
+         */
+        protected abstract void setMaximizedColumnProperties();
+
+        /**
+         * Sets the hidden columns for maximized-state.
+         */
+        protected abstract void setMaximizedHiddenColumns();
+
+        /**
+         * Sets additional headers for maximized-state.
+         */
+        protected abstract void setMaximizedHeaders();
+
+        /**
+         * Sets column expand ratio for maximized-state.
+         */
+        protected abstract void setMaximizedColumnExpandRatio();
+    }
+
+    /**
+     * Grids that are used in conjunction with
+     * {@link GeneratedPropertyContainer}, might use
+     * {@link AbstractGeneratedPropertySupport} to get type-save access to the
+     * raw container as well as the decorated container.
+     */
+    protected abstract class AbstractGeneratedPropertySupport {
+
+        /**
+         * Gives type-save access to the wrapper container the grid works on.
+         * This wrapper container attaches generated properties that are not
+         * part of the raw container that encapsulates the database access.
+         *
+         * @return decorated container that includes the generated properties as
+         *         well as the native properties.
+         */
+        public abstract GeneratedPropertyContainer getDecoratedContainer();
+
+        /**
+         * Gives type-save access to the wrapped container that binds to the
+         * database. The grid does not work directly with this container but
+         * with a container that wraps and decorates it with generated
+         * properties.
+         *
+         * @return raw container that gives access to the native properties.
+         */
+        public abstract T getRawContainer();
+
+        /**
+         * Adds the generated properties to the decorated container. Each
+         * generated property has to be associated with a property generator
+         * that is capable to calculate the value of the generated (aka virtual)
+         * property.
+         *
+         * @return decorated container that includes the generated properties
+         */
+        protected abstract GeneratedPropertyContainer addGeneratedContainerProperties();
+
+        /**
+         * Decorates the raw-container by wrapping it.
+         *
+         * @param container
+         *            raw-container to be wrapped.
+         * @return decorated container.
+         */
+        protected GeneratedPropertyContainer decorate(final T container) {
+            return new GeneratedPropertyContainer(container);
+        }
+    }
+
+    /**
+     * Support for single selection on the grid.
+     */
+    protected class SingleSelectionSupport {
+        private boolean enabled;
+
+        public SingleSelectionSupport(final boolean enabled) {
+            if (enabled) {
```

Comment:
Why do we need a specific selection support? 
Therefore there is a `selection model` and a `selection model`in Vaadin Grid. In my opinion the boolean is not necessary.

See https://vaadin.com/docs/-/part/framework/components/components-grid.html